core/src/main/java/org/openconnectors/connect/SinkConnector.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.openconnectors.connect;
+
+import org.openconnectors.util.SinkConnectorContext;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+
+public abstract class SinkConnector<T extends SinkConnectorContext, U> extends Connector<T> {
+
+    public abstract CompletableFuture<Void> publish(final Collection<U> messages);
```

Comment:
Collection isn't ordered. Perhaps List would make more sense? @aahmed-se

Reply:
I wouldn't want to restrict is at this level of the api, since not everyone connector will need strict ordering, List would make sense at a level above this, say an ordered sink connector

Reply:
:+1: